### PR TITLE
[v2.9] Fix/42823

### PR DESCRIPTION
**Pull request by gehrkefc**
## EDIT 1:
This PR was initially intended to bump to steve to a particular commit hash containing the fix for issue 42823 in release/v2.9 branch. But then an integration test started to fail. Then we've worked to fix it, and in the meanwhile someone updated the base branch with a most up to date commit hash from steve which already contains this bump. So, we just have the integration test fix now.

## Problem
Differences between create/update in ClusterRoleTemplateBindings
 
## Solution
Added the same logic in both

### CODE REVIEW

**Review by MbolotSuse — APPROVED** [review]
LGTM once the CI passes.

**Review by maxsokolovsky — APPROVED** [review]

**Review by joshmeranda — APPROVED** [review]

File: tests/v2/integration/catalogv2/cluster_repo_test.go [hunk 1]
```diff
@@ -817,6 +817,9 @@ func (c *ClusterRepoTestSuite) testClusterRepoRetries(params ClusterRepoParams)
 	})
 	require.NoError(c.T(), err)
 
+	cr, err = c.catalogClient.ClusterRepos().Get(context.TODO(), params.Name, metav1.GetOptions{})
```
Comment (diogoasouza) [inline tests/v2/integration/catalogv2/cluster_repo_test.go:820]: Why is this necessary?

Comment (gehrkefc) [reply tests/v2/integration/catalogv2/cluster_repo_test.go]: Just removed it. Thanks for the feedback!

**Review by diogoasouza — APPROVED** [review]

**Review by prachidamle — APPROVED** [review]
